### ISSUE TITLE
Configure DNS resolution on the compute node

### ISSUE THREAD

**Issue by anderbubble** [warewulf/warewulf#922]
In the past, resolv.conf was copied from the host into the the node image during container import. However, this has recently been replaced by read-only bind mounts.

We should add resolv.conf back into the container image by way of the wwinit overlay, just importing the contents of the file from the host by default.

**Comment by griznog** [comment]
The need for a `resolv.conf` in an overlay can be eliminated completely and produce a more robust `resolv.conf` along the way by setting `DNS[123]`, `SEARCH` and `DOMAIN` in the network config, both the `network` and `NetworkManager` services create this file from DHCP responses or static entries if configured in the interface configs.  Worse, `NetworkManager` is more than happy to overwrite `resolv.conf` with brokenness if given half a chance an mild encouragement anyway.

This is important to clusters with multihomed hosts who may not want all nodes to use the same DNS, e.g. login nodes with an external interface may want to point externally for DNS name resolution. For those of us using DNSMasq and getting DNS views for free by setting the nameservers to a specific interface on the master need to set up `resolv.conf` accordingly. 

Even as a simple default Just Works(tm) setup, adding DNS, SEARCH and DOMAIN to the network config and templates is much more robust and understandable than blindly copying from the master and hoping that works out downstream.

**Comment by anderbubble** [comment]
I agree with this for sure, but supporting `DNS`/`SEARCH`/`DOMAIN` in the network config will require code changes and would be a new feature, whereas populating `resolv.conf` from the head node as an overlay just fixes a regression. So my inclination is to do the `resolv.conf` thing for now and then add a feature request for 4.6 to add DNS/SEARCH/DOMAIN to the network config, potentially as part of breaking network config out from nodes.

**Comment by griznog** [comment]
This is yet another example of something that becomes much cleaner and easier if/when we move away from actual data structures for these to just managing a collection of YAML data with some suggested schemas that map to the official overlays/templates and "normal" deployments.

**Comment by mslacken** [comment]
This is not a bug, as the `resolv.conf` is created by e.g. NetworkManager or wicked during the network initialization. So having it in `wwinit` will create more headache as it solves.

**Comment by anderbubble** [comment]
It's at least a regression from previous behavior, though, since the way it used to work dns "just worked" by way of it being copied in from the compute image, if accidentally.

I like @griznog's suggestion of doing this with DNS settings in the network configuration scripts; but that doesn't appear to be supported, for example, at https://github.com/hpcng/warewulf/blob/development/overlays/wwinit/etc/sysconfig/network-scripts/ifcfg.ww.

**Comment by griznog** [comment]
I've been slowly moving all my network config into netdev tags and away from the data structure fields in parallel with working on an `ifcfg.ww` template that supports bonding, vlans and bridges (and stacking of those as needed). A profile contains all the defaults and effectively the schema for the netdev tags. If/when that is presentable I'll post an example  as a proof-of-concept. It's pretty crude at the moment because of the limitations imposed by our current nodeprofile + node config merging. 

The UI is simplified with this approach, as there is no need for any special `wwctl` command flags for network device configs, just need to manage tags as required for a given interface.  Without a fixed set of field names, the tag keys are no longer distro- or tool- specific so in my case  I can have a `service-network` overlay which handles old school *EL network type configs and a `service-NetworkManager` overlay for handling my future NetworkManger configuration, with flexibility in the tag key names to be able to support both during the transition. The only hard requirement for a netdev then is the `hwaddr` as `warewulfd` needs to know how to identify the node so this has to be set for all possible provisioning interfaces on a given node.

If/when we ever advance to being able to nest data structures in tags and nodeprofile/node config merging is smarter, this become a really powerful tool for configuring stuff via overlay.

**Comment by anderbubble** [comment]
New plan: we'll move this to 4.6 and incorporate it into new support for networks as a first-class entity. We can then support one or more DNS entries for the network itself.

**Comment by alansill** [comment]
Just a short comment to note that upstream has moved away from manual maintenance of resolv.conf in favor of systemd-resolved.
